File: 1st/1st.rs
use std::fs::File;
use std::io::{BufReader, BufRead};

fn main() {
    let mut file = File::open("in.txt").unwrap();
    let mut lines = BufReader::new(file).lines();
    let mut totals = vec![];
    let mut total = 0;
    for line in lines {
        match line.unwrap().as_str() {
            "" => {
                totals.push(total);
                total = 0;
            },
            num => {
                let shit: i32 = num.parse().unwrap();
                total += shit;
            }
        }
    }
    println!("{:?}", totals.iter().max());
}

File: 4th/1st.py
score = 0
with open("in.txt") as f:
    for row in f:
        rA, rB = row.split(",")
        lA, rA = rA.split('-')
        lB, rB = rB.split('-')
        lA, rA, lB, rB = map(lambda x: int(x), [lA, rA, lB, rB])
        if (lA >= lB and rA <= rB) or(lA <= lB and rA >= rB):
            score += 1
    print(score)
